src/plans/conjugate_gradient_plan.jl

```diff
@@ -1,3 +1,19 @@
+
+struct DirectionUpdateRuleStorage{TC<:DirectionUpdateRule,TStorage<:StoreStateAction}
+    coefficient::TC
+    storage::TStorage
+end
+
+function DirectionUpdateRuleStorage(
```

Comment:
maybe we could unify this to use keywords for p and x as `init_p=` and `init_X=` similar to the storage itself?

Reply:
Sure, I will change that.

Reply:
Done.